### PR TITLE
Add Dropout1d module

### LINKED ISSUE #6442: [feature request] dropout1d
While there are dropout2d and dropout3d, there is no dropout1d. And it seems like FeatureDropout naturally supports NxCxL input, should be no effort?

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #79549
* __->__ #79545

Fixes #6442

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

File: torch/nn/functional.py [hunk 1]
```diff
@@ -1264,6 +1264,44 @@ def alpha_dropout(input: Tensor, p: float = 0.5, training: bool = False, inplace
     return _VF.alpha_dropout_(input, p, training) if inplace else _VF.alpha_dropout(input, p, training)
 
 
+def dropout1d(input: Tensor, p: float = 0.5, training: bool = True, inplace: bool = False) -> Tensor:
```
Comment (ngimel) [inline torch/nn/functional.py:1267]: should there be feature_dropout_nd function that factors out common code between 1d/2d/3d? Doesn't need to happen now

File: torch/nn/modules/dropout.py [hunk 2]
```diff
@@ -58,6 +58,48 @@ def forward(self, input: Tensor) -> Tensor:
         return F.dropout(input, self.p, self.training, self.inplace)
 
 
+class Dropout1d(_DropoutNd):
+    r"""Randomly zero out entire channels (a channel is a 1D feature map,
```
Comment (ngimel) [inline torch/nn/modules/dropout.py:62]: same, doc strings can probably be shared between 1d/2d/3d

**Review by albanD — APPROVED** [review]
:+1:

File: torch/nn/functional.py (hunk 1 above)
Comment (jbschlosser) [reply torch/nn/functional.py:1267]: flagged for later - this should definitely happen

File: torch/nn/modules/dropout.py (hunk 2 above)
Comment (jbschlosser) [reply torch/nn/modules/dropout.py:62]: also flagged
